cob_twist_controller/CMakeLists.txt

```diff
@@ -25,8 +29,16 @@ include_directories(include ${catkin_INCLUDE_DIRS} ${EIGEN_INCLUDE_DIRS} ${oroco
 
 ## Build
 # Declare a cpp library
+add_library(damping_methods src/damping_methods/damping_base.cpp src/damping_methods/damping_manipulability.cpp src/damping_methods/damping_constant.cpp src/damping_methods/damping_builder.cpp)
+add_library(constraint_solvers ${SRC_CS_DIR}/constraint_solver_base.cpp ${SRC_CS_DIR}/unconstraint_solver.cpp ${SRC_CS_DIR}/wln_joint_limit_avoidance_solver.cpp ${SRC_CS_DIR}/weighted_least_norm_solver.cpp ${SRC_CS_FACT_DIR}/wln_joint_limit_avoidance_solver_factory.cpp ${SRC_CS_FACT_DIR}/solver_factory.cpp ${SRC_CS_FACT_DIR}/unconstraint_solver_factory.cpp ${SRC_CS_FACT_DIR}/weighted_least_norm_solver_factory.cpp ${SRC_C_DIR}/constraint_solver_factory_builder.cpp)
+target_link_libraries(constraint_solvers damping_methods)
```

Comment:
Agree. Too many files. 

Suggestion: 
Providing a file for the interface definitions (.h).
Providing a file for the abstract implementation of the interface (.cpp).
One header for all implementations.
But still several cpp files for the classes to separate implementation so it will keep up organized and clean. 

I am afraid of mixing the implementations in one cpp file e.g. for the solvers.